client/consensus/common/src/parachain_consensus.rs

```diff
@@ -27,15 +26,25 @@ use sp_runtime::{
 	traits::{Block as BlockT, Header as HeaderT},
 };
 
+use cumulus_primitives_core::{RecoveryDelay, RecoveryKind, RecoveryRequest};
+use cumulus_relay_chain_interface::{RelayChainInterface, RelayChainResult};
+
 use polkadot_primitives::v2::{Hash as PHash, Id as ParaId, OccupiedCoreAssumption};
 
 use codec::Decode;
-use futures::{select, FutureExt, Stream, StreamExt};
+use futures::{channel::mpsc::Sender, select, FutureExt, Stream, StreamExt};
 
-use std::{pin::Pin, sync::Arc};
+use std::{pin::Pin, sync::Arc, time::Duration};
 
 const LOG_TARGET: &str = "cumulus-consensus";
 
+// Delay range to trigger explicit requests.
+// The chosen value doesn't have any special meaning, a random delay within the order of
+// seconds in practice should be a good enough to allow a quick recovery without DOSing
+// the relay chain.
+const RECOVERY_DELAY: RecoveryDelay =
+	RecoveryDelay { min: Duration::ZERO, max: Duration::from_secs(30) };
```

Comment:
The "automatic" recovery requests triggered when the relay chain announces a candidate are submitted according to the values specified in the `service` crate.

https://github.com/paritytech/cumulus/blob/e7660ea44908a183964f69e19019f1ccb4d250ec/client/service/src/lib.rs#L128

https://github.com/paritytech/cumulus/blob/e7660ea44908a183964f69e19019f1ccb4d250ec/client/service/src/lib.rs#L219

As can be seen the values to be used are different depending if the node is a full-node or a collator.

---

This type of recovery here instead I think that should be independent of the type of node as we require the value immediately and the delay is passed only to prevent DOS-ing the relay chain with the same request from all the nodes

The value used here is pretty arbitrary but good enough. I.e. a random value that goes from 0 to 30 seconds...
Maybe we can lower it?